### DISCUSSION TITLE
How to get relation id when creating data

### DISCUSSION THREAD

**Discussion by Majestic-Fire** [pocketbase/pocketbase#4302] [Q&A] [answered]
| Collection 1 | Equipment    |
| ----------- | ----------- |
| id      | given |
| #TagID  | number |

| Collection 2 | EquipmentLogging    |
| ----------- | ----------- |
| id      | given |
| Equipment  | relation(Equipment) |
| Status| string|

```ts
        const data = {
            Equipment: form.data.equipmentId,   // How can i get the relation id  by #tagID
            Status: form.data.status,
         };

        const record = await pb.collection('EquipmentLogging').create(data);
```

---
for example
| Collection 1 | Equipment    |
| ----------- | ----------- |
| id      | z5e6kvii1hk9v3x |
| #TagID  | 3 |

| Collection 2 | EquipmentLogging    |
| ----------- | ----------- |
| id      | vg9ycr3d8opagfc |
| Equipment  | z5e6kvii1hk9v3x |
| Status| Pass |

How can i get  `z5e6kvii1hk9v3x` from #TagID = 3 with JS SDK

**Comment by ganigeorgiev** [comment]
The tables in your description don't seem to correspond to the shown code sample and I don't understand what you are trying to do.

For expanding relations, please refer to https://pocketbase.io/docs/working-with-relations/.

**Reply by Majestic-Fire** [reply]
```
import PocketBase from 'pocketbase';

const pb = new PocketBase('http://127.0.0.1:8090');

...

const post = await pb.collection('posts').create({
    'title': 'Lorem ipsum...',
    'tags':  ['TAG_ID1', 'TAG_ID2'],
});
```

As 'id' is auto generate, i would like to how to retrive the 'TAG_ID' , for example, by its tag name. Thanks in advance!

Collection: Tags
| 🗝id | name |
| ----------- | ----------- |
| z5e6kvii1hk9v3x | topic1 |
| haw8l8uo1xo1kaf| topic2 |

From,    
```
const post = await pb.collection('posts').create({
    'title': 'Lorem ipsum...',
    'tags':  ['z5e6kvii1hk9v3x', 'haw8l8uo1xo1kaf'],
});
```
to ,
```    
const post = await pb.collection('posts').create({
    'title': 'Lorem ipsum...',
    'tags':  ["topic1", "topic2"],
});
```

**Reply by ganigeorgiev** [reply, marked as answer]
Relation are associated only via the id of the record(s) and you need to fetch the related record(s) upfront. There is no other way around this (at least not out of the box).

One possible example could be to use the `getFullList`/`getFirstListItem`:

```js
const tags = await pb.collection("tags").getFullList({ filter: 'name = "topic1" || name = "topic2"' });

const post = await pb.collection('posts').create({
    'title': 'Lorem ipsum...',
    'tags':  tags.map((t) => t.id),
});
```
